include/motor-control/simulation/motor_interrupt_driver.hpp

```diff
@@ -38,7 +38,7 @@ class MotorInterruptDriver {
                         "%d, duration %ld",
                         move.group_id, move.seq_id, move.duration);
                     do {
-                        if (queue.peek(&move, 0)) {
+                        if (queue.try_read(&move, 0)) {
```

Comment:
Pretty sure this wasn't a bug, the interrupt handler should have a reference to the same queue as this class and it manages reading reading messages as they come in (all handled in the `pulse()` function)

Reply:
Yea I got confused because consecutive calls to execute_move_group re-ran all the moves again but in the monorepo hardware controller, we always clear the move groups before starting a new one